### PR TITLE
Fix/GitHub 1021/segments for edge identity

### LINKED ISSUE #1021: /api/v1/projects/{project_pk}/segments/ should support edge identity as well
https://sentry.io/organizations/flagsmith/issues/3190345735/?environment=staging&project=5544478&query=is%3Aunresolved&statsPeriod=24h

### PR DESCRIPTION
**Pull request by gagantrivedi**

### CODE REVIEW

File: api/environments/dynamodb/dynamodb_wrapper.py [hunk 1]
```diff
@@ -110,3 +110,9 @@ def write_environments(self, environments: Iterable[Environment]):
         with self._table.batch_writer() as writer:
             for environment in environments:
                 writer.put_item(Item=build_environment_document(environment))
+
+    def get_item(self, api_key: str) -> dict:
+        try:
+            return self._table.get_item(Key={"api_key": api_key})["Item"]
+        except KeyError as e:
```
Comment (matthewelwell) [inline api/environments/dynamodb/dynamodb_wrapper.py:133]: Is this correct? I have a feeling that we changed something recently to inspect whether the `"Item"` attribute was `None`?

File: api/segments/views.py [hunk 2]
```diff
@@ -42,9 +48,24 @@ def get_queryset(self):
...
+                identity = Identity.objects.get(pk=identity_pk)
+                segment_ids = [segment.id for segment in identity.get_segments()]
+            else:
+                segment_ids = _get_segment_ids_for_edge_identity(identity_pk)
+            queryset = queryset.filter(id__in=segment_ids)
 
         return queryset
+
+
+def _get_segment_ids_for_edge_identity(identity_pk: str) -> list:
```
Comment (matthewelwell) [inline api/segments/views.py]: It feels like this method would be better off living somewhere else so we can unit test it separately. 

Could we add it directly to the `Identity` model? Or perhaps to a dynamo module / wrapper?

File: api/environments/dynamodb/dynamodb_wrapper.py (hunk 1 above)
Comment (gagantrivedi) [reply api/environments/dynamodb/dynamodb_wrapper.py:133]: Yeah, that was around query Item returning empty list

File: api/segments/views.py (hunk 2 above)
Comment (gagantrivedi) [reply api/segments/views.py]: Moved to dynamo identity wrapper

File: api/environments/dynamodb/dynamodb_wrapper.py [hunk 3]
```diff
@@ -102,6 +106,18 @@ def search_items_with_identifier(
             query_kwargs.update(ExclusiveStartKey=start_key)
         return self.query_items(**query_kwargs)
 
+    def get_segmenent_ids(self, identity_pk: str) -> list:
```
Comment (matthewelwell) [inline api/environments/dynamodb/dynamodb_wrapper.py]: Uh oh! 

```suggestion
    def get_segment_ids(self, identity_pk: str) -> list:
```

**Review by matthewelwell — COMMENTED** [review]
Looks good apart from the typo.

**Review by matthewelwell — APPROVED** [review]

**Pull request merged**